### PR TITLE
[PR-9208] Async liquid docs

### PR DESCRIPTION
**Pull request by antonpaisov**
Related PR: https://github.com/talkable/talkable/pull/4606

## Demo

http://void-docs.talkable.com/

## Related Stories

[](https://talkable.atlassian.net/browse/PR-9208)

### CODE REVIEW

File: source/campaigns/editor/filters.rst [hunk 1]
```diff
@@ -665,3 +665,84 @@ Takes a hash and returns the array of its values.
...
 Returns ``[{ad incentive config}, {fr incentive config}]``
+
+|hr|
+
+async\_rendering
+................
+
+Stores liquid partial that can be later rendered via ``Talkable.loadLiquid`` function.
+
+Expample 1
```
Comment (zhuravel) [inline source/campaigns/editor/filters.rst]: Expample

File: source/campaigns/editor/filters.rst [hunk 2]
```diff
@@ -665,3 +665,84 @@ Takes a hash and returns the array of its values.
...
+      console.log(rendered_template);
+    });
+
+Rendered liquid
+
+.. code-block:: text
+
+            "template for rendering with data advocate@example.com"
+
+Expample 2
```
Comment (zhuravel) [inline source/campaigns/editor/filters.rst]: Expample

File: source/campaigns/editor/filters.rst [hunk 3]
```diff
@@ -665,3 +665,84 @@ Takes a hash and returns the array of its values.
...
+    console.log(rendered_data);
+  });
+
+Rendered liquid
+
+.. code-block:: text
+
+  {some_key: "$420", other_key: 12}
+
+Expample 3
```
Comment (zhuravel) [inline source/campaigns/editor/filters.rst]: Expample

File: source/campaigns/editor/filters.rst [hunk 4]
```diff
@@ -665,3 +665,84 @@ Takes a hash and returns the array of its values.
...
+~~~~~~~~~~
+
+Liquid
+
+.. code-block:: liquid
+
+  {% capture some_liquid_block %}
+  {% raw %}
+  Whatever liquid you would live to have
```
Comment (zhuravel) [inline source/campaigns/editor/filters.rst]: liquid → Liquid
live → like

File: source/campaigns/editor/filters.rst [hunk 5]
```diff
@@ -665,3 +665,84 @@ Takes a hash and returns the array of its values.
...
+  {% endraw %}
+  {% endcapture %}
+  {{ 'async_liquid_reference' | async_rendering: some_liquid_block }}
+
+Javascript API call
```
Comment (zhuravel) [inline source/campaigns/editor/filters.rst]: Javascript → JavaScript

File: source/campaigns/editor/filters.rst [hunk 6]
```diff
@@ -665,3 +665,84 @@ Takes a hash and returns the array of its values.
    {% assign incentive_configs = incentives | values %}
 
 Returns ``[{ad incentive config}, {fr incentive config}]``
+
+|hr|
+
+async\_rendering
+................
+
+Stores liquid partial that can be later rendered via ``Talkable.loadLiquid`` function.
```
Comment (zhuravel) [inline source/campaigns/editor/filters.rst]: liquid → Liquid

File: source/campaigns/editor/filters.rst [hunk 7]
```diff
@@ -665,3 +665,84 @@ Takes a hash and returns the array of its values.
...
+Expample 1
+~~~~~~~~~~
+
+Liquid
+
+.. code-block:: liquid
+
+   {{ "async_advocate_info" | async_rendering: 'template for rendering with data [[ advocate_info.email ]]' }}
+
+Javascript API call
```
Comment (zhuravel) [inline source/campaigns/editor/filters.rst]: Javascript → JavaScript

File: source/campaigns/editor/filters.rst [hunk 8]
```diff
@@ -665,3 +665,84 @@ Takes a hash and returns the array of its values.
...
+
+.. code-block:: javascript
+
+    Talkable.loadLiquid("async_advocate_info", function(rendered_template) {
+      console.log(rendered_template);
+    });
+
+Rendered liquid
```
Comment (zhuravel) [inline source/campaigns/editor/filters.rst]: liquid → Liquid

File: source/campaigns/editor/filters.rst [hunk 9]
```diff
@@ -665,3 +665,84 @@ Takes a hash and returns the array of its values.
...
+
+.. code-block:: text
+
+            "template for rendering with data advocate@example.com"
```
Comment (zhuravel) [inline source/campaigns/editor/filters.rst]: Too many spaces.

File: source/campaigns/editor/filters.rst [hunk 10]
```diff
@@ -665,3 +665,84 @@ Takes a hash and returns the array of its values.
...
+Expample 2
+~~~~~~~~~~
+
+Liquid
+
+.. code-block:: liquid
+
+   {{  'async_dashboard_data' | async_rendering: some_key: "[[ dashboard.possible_rewards | money: strip_insignificant_zeros: true]]",  other_key: "[[ friend_info.purchases_count ]]" }}
+
+Javascript API call
```
Comment (zhuravel) [inline source/campaigns/editor/filters.rst]: Javascript → JavaScript

File: source/campaigns/editor/filters.rst [hunk 11]
```diff
@@ -665,3 +665,84 @@ Takes a hash and returns the array of its values.
...
+            "template for rendering with data advocate@example.com"
+
+Expample 2
+~~~~~~~~~~
+
+Liquid
+
+.. code-block:: liquid
+
+   {{  'async_dashboard_data' | async_rendering: some_key: "[[ dashboard.possible_rewards | money: strip_insignificant_zeros: true]]",  other_key: "[[ friend_info.purchases_count ]]" }}
```
Comment (zhuravel) [inline source/campaigns/editor/filters.rst]: Extra space before `other_key`.

File: source/campaigns/editor/filters.rst [hunk 12]
```diff
@@ -665,3 +665,84 @@ Takes a hash and returns the array of its values.
...
+
+Javascript API call
+
+.. code-block:: javascript
+
+  Talkable.loadLiquid("async_dashboard_data", function(rendered_data) {
+    console.log(rendered_data);
+  });
+
+Rendered liquid
```
Comment (zhuravel) [inline source/campaigns/editor/filters.rst]: liquid → Liquid

File: source/campaigns/editor/filters.rst [hunk 13]
```diff
@@ -665,3 +665,84 @@ Takes a hash and returns the array of its values.
...
+
+Javascript API call
+
+.. code-block:: javascript
+
+  Talkable.loadLiquid("async_liquid_reference", function(rendered_template) {
+    console.log(rendered_template);
+  });
+
+Rendered liquid
```
Comment (zhuravel) [inline source/campaigns/editor/filters.rst]: liquid → Liquid

File: source/campaigns/editor/filters.rst [hunk 14]
```diff
@@ -665,3 +665,84 @@ Takes a hash and returns the array of its values.
...
+
+.. code-block:: text
+
+  "Whatever liquid you would live to have"
```
Comment (zhuravel) [inline source/campaigns/editor/filters.rst]: liquid → Liquid
live → like

**Comment by antonpaisov** [comment]
@zhuravel do I write about the error callback somewhere here?

File: source/campaigns/editor/filters.rst [hunk 15]
```diff
@@ -665,3 +665,84 @@ Takes a hash and returns the array of its values.
...
+
+  Talkable.loadLiquid("async_liquid_reference", function(rendered_template) {
+    console.log(rendered_template);
+  });
+
+Rendered Liquid
+
+.. code-block:: text
+
+  "Whatever Liquid you would live to have"
```
Comment (zhuravel) [inline source/campaigns/editor/filters.rst]: live → like

File: source/campaigns/editor/filters.rst [hunk 16]
```diff
@@ -665,3 +665,84 @@ Takes a hash and returns the array of its values.
...
+            "template for rendering with data advocate@example.com"
+
+Example 2
+~~~~~~~~~
+
+Liquid
+
+.. code-block:: liquid
+
+   {{  'async_dashboard_data' | async_rendering: some_key: "[[ dashboard.possible_rewards | money: strip_insignificant_zeros: true]]", other_key: "[[ friend_info.purchases_count ]]" }}
```
Comment (zhuravel) [inline source/campaigns/editor/filters.rst]: * Extra space after `{{`.
* Missing space in `true]]`.

File: source/campaigns/editor/filters.rst (hunk 6 above)
Comment (zhuravel) [reply source/campaigns/editor/filters.rst]: Up

**Comment by zhuravel** [comment]
@antonpaisov Regarding mentioning the error callback in docs — I don’t think this is required. The docs cover liquid filters, and error callback is not directly related to liquid filter, it’s a part of our JS lib.

File: source/campaigns/editor/filters.rst [hunk 17]
```diff
@@ -665,3 +665,84 @@ Takes a hash and returns the array of its values.
...
+
+  Talkable.loadLiquid("async_dashboard_data", function(rendered_data) {
+    console.log(rendered_data);
+  });
+
+Rendered Liquid
+
+.. code-block:: text
+
+  {some_key: "$420", other_key: 12}
```
Comment (zhuravel) [inline source/campaigns/editor/filters.rst]: I think rendered Liquid will be a valid JSON here. Correct me if I’m wrong.
``` suggestion
  {"some_key":"$420","other_key":12}

**Pull request merged**